#include <iostream>
#include <stdexcept>

int factorial(int n) {
    if (n < 0)
        throw std::invalid_argument("Factorial is not defined for negative numbers.");
    if (n == 0 || n == 1)
        return 1;
    return n * factorial(n - 1);
}

int main() {
    int num;
    std::cout << "Enter a positive number to calculate its factorial: ";
    std::cin >> num;

    try {
        int result = factorial(num);
        std::cout << "Factorial of " << num << " is " << result << std::endl;
    } catch(const std::invalid_argument& e) {
        std::cerr << "Error: " << e.what() << std::endl;
    }

    return 0;
}
